nixos/modules/services/databases/postgresql.nix

```diff
@@ -622,6 +622,41 @@ in
             TimeoutSec = 120;
 
             ExecStart = "${postgresql}/bin/postgres";
+
+            # Hardening
+            CapabilityBoundingSet = [ "" ];
+            DevicePolicy = "closed";
+            PrivateTmp = false; #breaks wal-receiver test
+            ProtectHome = true;
+            ProtectSystem = "strict";
+            MemoryDenyWriteExecute = true;
+            NoNewPrivileges = true;
+            LockPersonality = true;
+            PrivateDevices = true;
+            PrivateMounts = false; # breaks wal-receiver test
+            ProcSubset = "pid";
+            ProtectClock = true;
+            ProtectControlGroups = true;
+            ProtectHostname = true;
+            ProtectKernelLogs = true;
+            ProtectKernelModules = true;
+            ProtectKernelTunables = true;
+            ProtectProc = "invisible";
+            RemoveIPC = true;
+            RestrictAddressFamilies = [
+              "AF_UNIX"
+              "AF_INET"
+              "AF_INET6"
+            ];
+            RestrictNamespaces = true;
+            RestrictRealtime = true;
+            RestrictSUIDSGID = true;
+            SystemCallArchitectures = "native";
+            SystemCallFilter = [
+              "@system-service"
+              "~@privileged @resources"
+            ];
+            UMask = "0077";
```

Comment:
I would use 137 here. I don't think it's a problem when the postgres group has read access.

Reply:
I would be fine with 0027, so `u=rwx,g=rx,o=`. WDYT?

Reply:
I like the idea from commit 64565d6 making this conditional on `groupAccessAvailable`.

However, why would the Postgres files need the execute bit? The data files don't need that. The only executables shipped by Postgres should be in the Nix store. I suggest to drop it for both, user and group members, but I think it should definitely be dropped for the group.

Reply:
It is already tied to that in the latest changeset. The executable bit is for directory access.

Reply:
I thought there is something different handling directories, but it seems not. I'm fine with it.